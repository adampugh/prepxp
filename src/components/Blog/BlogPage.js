import React, { Component } from "react";

import Navbar from "../UI/navbarLoggedIn";
import BlogBanner from "../Blog/blogBanner";
import BlogGrid from "./blogGrid";
import Footer from "../UI/footer";

class BlogPage extends Component {
    state = {
        posts: [
            {
                title: "5 Tips to get things done",
                text: "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
                img: "http://via.placeholder.com/300x300"
            }, {
                title: "Pass the interview with these answers",
                text: "sf sdfg dfasd df dg asdf sdf adsf sadf",
                img: "http://via.placeholder.com/300x300"
            }, {
                title: "Top 10 most common questions",
                text: "sf sdfg dfasd df dg asdf sdf adsf sadf",
                img: "http://via.placeholder.com/300x300"
            }, {
                title: "What Google and employers are looking for",
                text: "sf sdfg dfasd df dg asdf sdf adsf sadf",
                img: "http://via.placeholder.com/300x300"
            },
        ]
    }



    render() {
        return (
            <div>
                <Navbar />
                <BlogBanner />
                <BlogGrid posts={this.state.posts}/>
                <Footer />
            </div>
        );
    }
}

export default BlogPage;