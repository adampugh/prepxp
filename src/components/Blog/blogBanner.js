import React from "react";

import blog from "../../assets/images/blog.png";

const blogBanner = () => (
    <div className="blogBanner">
        <div className="container blogBanner__grid">
            <div className="blogBanner__img">
                <img src={blog} alt="" />
            </div>
            <div className="blogBanner__text">
                <h1>Latest blog posts</h1>
                <p>Read articles from industry leaders on how to best prepare for 
                interviews and put your best foot forward. 
                Read articles from industry leaders on how to best prepare for 
                interviews and put your best foot forward.
                </p>
                <p>Read articles from industry leaders on how to best prepare for 
                interviews and put your best foot forward. 
                Read articles from industry leaders on how to best prepare for 
                interviews and put your best foot forward.
                </p>
            </div>
        </div>
    </div>
);

export default blogBanner;
