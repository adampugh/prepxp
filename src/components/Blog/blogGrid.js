import React from "react";
import { Link } from "react-router-dom";


import BlogPanel from "./blogPanel";

const blogGrid = (props) => (
    <div className="blog__background">
        <div className="container">
            <div className="blog__grid">
                {props.posts.map((post, index) => (
                    <Link key={post.title + index} to={{pathname: `/blog/${post.title.split(" ").join("-")}`, state:{post}}}>
                        <BlogPanel post={post}/>
                    </Link>
                ))}
            </div>
        </div>
    </div>
);

export default blogGrid;