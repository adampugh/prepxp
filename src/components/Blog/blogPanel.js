import React from "react";

const blogPanel = (props) => (
    <div className="blog__post">
        
        <div className="blog__post__text">
            <p>{props.post.date}</p>
            <h2>{props.post.title}</h2>
            <hr />
            <p className="blog__post__text__author">{props.post.author}</p>
            <div className="blog__post__text__comments">
                <p>{props.comments.length}<i className="fas fa-comment"></i></p>
            </div>
        </div>
        <div 
            className="blog__post__img"
            style={{backgroundImage:`url(${props.post.img})`}}>
            <div className="blog__post__img__overlay">
            <p>Read <i className="fas fa-arrow-right"></i></p>
            </div>
        </div>
        
    </div>
);

export default blogPanel;
