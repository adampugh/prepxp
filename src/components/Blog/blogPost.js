import React from "react";

const blogPost = (props) => (
    <div className="blog__post">
        <img src={props.post.img} alt="blog post" />
        <div className="blog__post__text">
            <h2>{props.post.title}</h2>
            <p>{props.post.text}</p>
        </div>
    </div>
);

export default blogPost;