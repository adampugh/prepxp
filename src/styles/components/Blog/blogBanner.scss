.blogBanner {
    background: $purple-gradient, url("../assets/images/bg5.jpg") no-repeat center;
    background-size: cover;
    color: white;
    position: relative;
    padding-top: 40px;
    // margin-bottom: -10vw;
    // min-height: 634px;
    min-height: 534px;
    margin-bottom: -50px;
}
  
  .blogBanner__whiteElipse {
    clip-path: ellipse(73% 87% at 68% 94%);
    // background: $very-light-grey;
    background-color: $light-purple;
    // height: 280px;
    height: 180px;
    position: absolute;
    width: 100%;
    bottom: 0;
  }


.blogBanner__grid {
    display: grid;
    grid-template-columns: 1fr;
    padding: $padding-s 0;
}


.blogBanner__img {
    // justify-self: center;
    
    z-index: 100;
    // -ms-transform: rotate(30deg); /* IE 9 */
    // -webkit-transform: rotate(30deg); /* Safari */
    // transform: rotate(30deg);
    img {
        width: 100%;
        padding: 5px;
        box-sizing: border-box;
    }
}

.blogBanner__text {
    // background: white;
    // border-radius: 20px;
    // color: $off-grey;
    // align-self: center;
    // padding: $padding-s;
    margin-bottom: $padding-m;
    padding: $padding-s;
    padding-bottom: 0;
    color: white;
    z-index: 100;
    // @include boxshadow;
    h1 {
        @include header;
        font-size: 7em;
        // margin-bottom: $padding-s;
    }
    p {
        // @include header;
        // color: $off-grey;
        // padding-top: $padding-s;
        font-size: $font-m;
    }
}

@media (min-width: $desktop-breakpoint) {
    .blogBanner {
        // padding-bottom: $padding-s;
        // margin-bottom: -50px;
    }

    .blogBanner__grid {
        grid-template-columns: 1fr 1fr;
        padding: $padding-m 0;
        img {
            width: 100%;
            // margin-top: $padding-s;
            padding-top: 0;
            // margin-left: 100px;
            // margin: $padding-m 0;
        }
    }

    .blogBanner__text {
        margin-bottom: 0;
        // padding: $padding-s $padding-s 0 0;
        // padding: 0;
        padding-top: 0;
        // padding-right: $padding-s;
    }
}