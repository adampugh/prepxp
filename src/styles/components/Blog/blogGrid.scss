.blog__background {
    background-color: $very-light-grey;
    padding: $padding-m 0;
}

.blog__grid {
    display: grid;
    // padding: $padding-m 0;
    padding-bottom: $padding-m;
}

.blog__grid__error {
    background-color: $very-light-grey;
    padding: $padding-l 0;
    margin: 0;
    text-align: center;
}

// .blog__container {
//     hr {
//         width: 25px;
//     }
// }

.blog__title {
    @include header;
    color: $off-purple;
    text-align: center;
    padding: $padding-s 0;
    &::after {
        content: "";
        display: block;
        width: 50px;
        height: 0.125rem;
        margin: 1rem auto;
        background: $off-purple;
    }
}

@media (min-width: $desktop-breakpoint) {
    .blog__grid {
        grid-template-columns: 1fr 1fr;
    }
}