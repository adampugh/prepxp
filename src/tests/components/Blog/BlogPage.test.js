import React from "react";
import { shallow } from "enzyme";
import { BlogPage } from "../../../components/Blog/BlogPage";


describe("<BlogPage />", () => {
    // 1. should match snapshot
    // 2. should call fetchBlogPosts
    // 3. fetchBlogPosts should return an array async

    test("should match snapshot", () => {
        const wrapper = shallow(<BlogPage />);
        expect(wrapper).toMatchSnapshot();
    });
        
});


