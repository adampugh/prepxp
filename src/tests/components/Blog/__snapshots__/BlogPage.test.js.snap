// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPage /> should match snapshot 1`] = `
<div>
  <Connect(Navbar) />
  <blogBanner />
  <blogGrid
    allPosts={Array []}
    editorPosts={Array []}
    error={null}
    loading={true}
    posts={Array []}
  />
  <footer />
</div>
`;
